src/Library/demos/Launcher/main.js

```diff
@@ -0,0 +1,91 @@
+import Gtk from "gi://Gtk";
+import Gdk from "gi://Gdk";
+import Gio from "gi://Gio";
+import GLib from "gi://GLib";
+
+const launch_file = workbench.builder.get_object("launch_file");
+const file_details = workbench.builder.get_object("file_details");
+const file_location = workbench.builder.get_object("file_location");
+const change_file = workbench.builder.get_object("change_file");
+const uri_launch = workbench.builder.get_object("uri_launch");
+const uri_details = workbench.builder.get_object("uri_details");
+const change_uri = workbench.builder.get_object("change_uri");
+
+Gio._promisify(Gtk.FileLauncher.prototype, "launch", "launch_finish");
+Gio._promisify(
+  Gtk.FileLauncher.prototype,
+  "open_containing_folder",
+  "open_containing_folder_finish",
+);
+Gio._promisify(Gtk.FileDialog.prototype, "open", "open_finish");
+Gio._promisify(Gtk.UriLauncher.prototype, "launch", "launch_finish");
+
```

Comment:
All the code here great, let's just separate them into two chunks with a comment like:

```js
// File Launcher
// < launcher code here >

// URI Launcher
// < launcher code here >